reframe/core/meta.py

```diff
@@ -33,3 +35,27 @@ def __init__(cls, name, bases, namespace, **kwargs):
 
         hooks['post_setup'] = fn_with_deps + hooks.get('post_setup', [])
         cls._rfm_pipeline_hooks = hooks
+
+        cls._final_methods = {v.__name__ for v in namespace.values()
+                              if hasattr(v, '_final')}
+
+        # Add the final functions from its parents
+        cls._final_methods.update(*[b._final_methods for b in bases
+                                    if hasattr(b, '_final_methods')])
+
+        # Filter the methods that are defined in this class but are not final
+        for v in namespace.values():
+            if callable(v) and not hasattr(v, '_final'):
+                cls._final_methods.discard(v.__name__)
```

Comment:
Why this is needed? You want essentially to override the `final` keyword? I don't think we need that and it's not correct conceptually. Normally, the `final` keyword would have effect over all subclasses. We just want to insert an exception to that with the special test classes, so the exception should be true only for the current class and not for its subclasses.

Reply:
I will remove it then. I did it so that we are able to override the behavior of final for special classes.